### ISSUE TITLE
使用模型預測

### ISSUE THREAD

**Issue by Balabala125** [gaussic/text-classification-cnn-rnn#7]
您好，我使用該模型進行預測，我在您的基础上写了一个预测函数，但总感觉不对，想请您帮忙看一下可以吗？（实在没办法了，看了好久还是没发现哪里有问题，身边的同学也不太懂，所以。。。。）麻烦您指点一下，真的非常感谢！！！！

下面这个是我写的预测函数：
#预测函数
def prediction():
    input_x = tf.placeholder(tf.int32, name='input_x')
    x_predict = preocess_file(data_path='data/cnews/', seq_length=600)
    saver = tf.train.Saver()

    with tf.Session() as sess:
        saver.restore(sess, './model')

        y = tf.get_collection('pred_network')[0]

        sess.run(input_x,feed_dict={input_x:x_predict})
    print(y)

在您的代码的基础上，我还在模型的定义部分的分类器部分加了一点代码，如下：（就是最后两行代码）
# 分类器
            self.logits = tf.layers.dense(fc, self.config.num_classes,
                name='fc2')
            self.pred_y = tf.nn.softmax(self.logits)
            predict = tf.argmax(self.pred_y, 1)
            tf.add_to_collection('pred_network', predict)

然后运行的结果是：（就是print(y)的结果）
Tensor("score/ArgMax:0", shape=(?,), dtype=int64)

**Comment by gaussic** [comment]
你好，这个问题主要看对TF的理解有多少。这样获得的y仍然是一个Tensor，是还没有运行的，所以无法取到数据，要用session.run()运行。

此外，我将我的代码重构了以下，将CNN和RNN的训练与测试分离到了两个文件，方便大家查看。

另外，添加了saver，以及预测用的测试函数，具体内容请看代码，希望能解决你的一些问题。

**Comment by Balabala125** [comment]
好的，谢谢您，我再仔细的看一下
